backends/ebpf/psa/ebpfPsaDeparser.cpp

```diff
@@ -33,6 +33,33 @@ void DeparserBodyTranslatorPSA::processFunction(const P4::ExternFunction *functi
     }
 }
 
+void DeparserBodyTranslatorPSA::processMethod(const P4::ExternMethod *method) {
+    if (method->method->name.name == "pack") {
+        // Emit digest pack method
+        auto obj = method->object;
+        auto di = obj->to<IR::Declaration_Instance>();
+        cstring digestMapName = EBPFObject::externalName(di);
+        auto arg = method->expr->arguments->front();
+        builder->appendFormat("bpf_map_push_elem(&%s, &", digestMapName);
+        this->visit(arg);
+        builder->appendFormat(", BPF_EXIST)");
+        return;
+    }
+
+    DeparserBodyTranslator::processMethod(method);
+}
+
+void EBPFDeparserPSA::emitDigestInstances(CodeBuilder* builder) const {
```

Comment:
I would name this function differently.
First, it is really about the control-plane API. Perhaps there should be a common prefix/suffix to indicate all functions that deal with control plane APIs. Second, it's really about creating the API which allows digest messages to be received. Perhaps something like "createDigestAPI"?

Reply:
We use `emitTableInstances` for generating instances of BPF maps implementing P4 tables. Why would we need to change this convention for Digests?

Reply:
I guess the way PSA implements digests makes this design choice reasonable.